File: ash/ash2/Dao/Hibernate/Tb_userDao.cs
using System;
using System.Collections;
using System.Linq;
using System.Web;
using Spring.Data.NHibernate;
using NHibernate;
using ash2.Dao.Interf;
using model;
using Spring.Data.NHibernate.Support;
using Spring.Transaction;
using Spring.Stereotype;
using Spring.Transaction.Interceptor;

namespace ash2.Dao.Hibernate
{
    //[Repository]
    public class Tb_userDao : HibernateDaoSupport, ITb_userDao
    {
        //private HibernateTemplate ht;
        //public ISessionFactory sf { set; get; }

        //private void setHibernateTemplate()
        //{
        //    if (null == ht)
        //    {
        //        ht = new HibernateTemplate(this.sf);
        //        ht.TemplateFlushMode = TemplateFlushMode.Auto;
        //        ht.CacheQueries = true;
        //    }
        //}

        public void delete(Tb_user model)
        {
            //sf.GetCurrentSession().Delete(tb_user);
            HibernateTemplate.Delete(model);
        }

        public Tb_user get(int id)
        {
            //setHibernateTemplate();
            return HibernateTemplate.Get(typeof(Tb_user), id) as Tb_user;
        }

        //[Transaction(Spring.Transaction.TransactionPropagation.Required, ReadOnly = false)]
        //[Transaction]
        public void save(Tb_user model)
        {
            //setHibernateTemplate();
            //sf.GetCurrentSession().Save(person);
            //ht.Save(person);
            //ISession s = sf.OpenSession();
            //ITransaction t = s.BeginTransaction();
            //s.FlushMode = FlushMode.Auto;
            //s.Save(person);
            //t.Commit();
            //HibernateTemplate.TemplateFlushMode = TemplateFlushMode.Auto;
            HibernateTemplate.Save(model);

            //ISession s = HibernateTemplate.SessionFactory.OpenSession();
            //ITransaction t = s.BeginTransaction();
            //s.Save(model);
            //t.Commit();
            //s.Close();
        }

        public void update(Tb_user model)
        {
            //setHibernateTemplate();
            //HibernateTemplate.Update(person);
            HibernateTemplate.Update(model);
        }

        public IList getAll()
        {
            //setHibernateTemplate();
            //return HibernateTemplate.Find("from Tb_user");
            return HibernateTemplate.Find("from Tb_user");
        }
    }
}

File: ssh/src/dao/hibernate/Tb_userDao.java
package dao.hibernate;

import java.util.List;

import model.test.Tb_user;

import org.hibernate.HibernateException;
import org.hibernate.Session;
import org.springframework.orm.hibernate3.HibernateCallback;
import org.springframework.orm.hibernate3.support.HibernateDaoSupport;

import dao.interf.ITb_userDao;

public class Tb_userDao extends HibernateDaoSupport implements ITb_userDao {
	// private HibernateTemplate ht;
	// private SessionFactory sessionFactory;
	//
	// private void setHibernateTemplate()
	// {
	// if (null == ht)
	// {
	// ht = new HibernateTemplate(this.sessionFactory);
	// }
	// }
	//
	// public void setSessionFactory(SessionFactory sessionFactory)
	// {
	// this.sessionFactory = sessionFactory;get
	// }

	public void delete(Tb_user model) {
		// setHibernateTemplate();
		// ht.delete(model);
		getHibernateTemplate().delete(model);
	}

	public void deleteById(int id) {
		// setHibernateTemplate();
		// ht.delete(model);
		Tb_user model = new Tb_user();
		model.setId(id);
		delete(model);
	}

	public Tb_user getById(int id) {
		// Session session =
		// getHibernateTemplate().getSessionFactory().openSession();
		// session.get(Tb_user.class, new Integer(id));
		return (Tb_user) getHibernateTemplate().get(Tb_user.class,
				new Integer(id));
	}

	public void save(Tb_user model) {
		getHibernateTemplate().save(model);
	}

	public void update(Tb_user model) {
		getHibernateTemplate().update(model);
	}

	@SuppressWarnings("unchecked")
	public List getAll() {
		// return getHibernateTemplate().find("from Tb_user");
		// http://kewb.javaeye.com/blog/128708
		return (List) getHibernateTemplate().execute(new HibernateCallback() {
			public Object doInHibernate(Session session)
					throws HibernateException {
				// List result =
				// session.createCriteria(Tb_user.class).add(Restrictions.like("name",
				// name+"%").list();
				// return result;
				return null;
			}
		});

	}
}
